MusicNotationKit/MusicNotationKit/Measure.swift

```diff
@@ -238,20 +208,8 @@ public struct Measure: ImmutableMeasure, Equatable {
         }
     }
 
-    private func tieStateForNoteIndex(_ noteIndex: Int, tupletIndex: Int?, inSet setIndex: Int) throws -> Tie? {
-        if let tupletIndex = tupletIndex {
-            guard let tuplet = notes[setIndex][noteIndex] as? Tuplet else {
-                assertionFailure("NoteCollection was not a Tuplet as expected")
-                throw MeasureError.internalError
-            }
-            return try tuplet.note(at: tupletIndex).tie
-        } else {
-            guard let note = notes[setIndex][noteIndex] as? Note else {
-                assertionFailure("NoteCollection was not a Note as expected")
-                throw MeasureError.internalError
-            }
-            return note.tie
-        }
+    private func tieStateForNoteIndex(_ index: Int, inSet setIndex: Int) throws -> Tie? {
```

Comment:
For Swift 3, change this to `private func tieState(forNoteIndex index: Int, inSet setIndex: Int) throws -> Tie?`. Normally you wouldn't even have `NoteIndex` in the first parameter, but I believe since we specifying what specific type of index, it is needed in this case.